### DISCUSSION TITLE
Can't upgrade, stuck at 0.5.3 with 4FUS/y25

### DISCUSSION THREAD

**Discussion by pultars** [roleoroleo/yi-hack-MStar#534] [Q&A] [answered]
I have been trying to upgrade my cam fw to the latest release but the upgrade doesn't seem to work. I have tried downgrading with multiple different versions as well, but that doesn't work either. I tried both putting the release file as y25_x.x.x.tgz in /tmp/sd and also taking out the SD card and extracting the home_y25 and sys_y25files into the same folder and then trying to upgrade through website but also by calling the cgi-bin/fw_upgrade.sh script, but none of these combinations seem to work. I tried have tried resetting the camera using the pinhole button, that didn't help with the upgrade either.

The upgrade process does create the .fw_upgrade folder with the files in it, and also overwrites the home_y25 and sys_25 files with the new release files but after the reboot, no love! Version is still at 0.5.3. Also, as an additional piece of information: I had upgraded to a beta release (https://github.com/roleoroleo/yi-hack-MStar/issues/531#issuecomment-1894363236) that @roleoroleo had released for fixing a bug I had encountered. Could this be a reason?

@roleoroleo had mentioned in some of the issues comments that the bootloader looks at a hash and decides to load or not, but I have no clue as to how to check these. I did turn on the debug and created the hack_debug.log file, but the file doesn't contain any clues to a hash compare and not liking it, etc. Only two sets of some starting services messages, PID list and mem status.

Any help would greatly appreciated...

Lastly, @roleoroleo, thanks again for all the work you have done on these projects!

**Comment by roleoroleo** [comment]
Are you sure that your cam is a y25?

The hash is different than the previous version and the beta you tried.
It must work.

**Reply by pultars** [reply]
The camera is a 4FUS with 4.2.0.0H_201909041620. (that is what the web ui shows)

I had mistakenly upgraded to a y23 on it at some point and the serial numbers had changed to random letters which made me realize that I had the wrong hack version. I then upgraded back to y25 0.5.3. Could this be a reason?

**Reply by roleoroleo** [reply]
No.
The upgrade procedure is handled by the bootloader.
And the bootloader is never modified.

Could you try with another sd card?

**Reply by pultars** [reply]
I actually have 4 cameras and the upgrade doesn't work on either. They all have separate SD cards. But I will try with some other SD card.

**Reply by roleoroleo** [reply]
You could try to downgrade to a previous version and then apply again the last version.

**Reply by pultars** [reply]
I tried a different SD card and that did not work.
I had tried downgrading previously and had not worked.
I tried downgrading with the different SD card, that didn't work.

What did work is that after extracting the home_y25 and sys_y25 files to the SD card, renaming them to home_y23 and sys_y23. Maybe the bootloader is now looking for y23 files after my mistake of using y23?

The status page reports y25 and 0.5.4 now.

Just to try, I tried downgrading with the regular procedure after managing to put 0.5.4 on there but it didn't downgrade.

**Reply by roleoroleo** [reply, marked as answer]
If your cam accepts home_y23 and sys_y23, it's a y23.
So, you have to use y23.

**Comment by pultars** [comment]
When I use y23, the serial number displayed is incorrect. Is this normal?

![Screenshot from 2024-01-24 18-47-58](https://github.com/roleoroleo/yi-hack-MStar/assets/24656144/e830a8d9-fc8a-4bd8-ac87-62bfbf10c28b)

**Comment by roleoroleo** [comment]
No, it's not normal.
Maybe a different revision of the yi firmware.
But the code is the same as y25.
Send me a copy of the file /tmp/mmap.info

**Comment by roleoroleo** [comment]
The offset of the serial number, is different.
I fixed it.
https://github.com/roleoroleo/yi-hack-MStar/commit/d54e74078002847c1c03a06115e4afa499c373d2

**Comment by pultars** [comment]
Thank you!
